### PR TITLE
DM-34986: build with pyproject.toml

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+urls = { documentation = "https://ts-ess-labjack.lsst.io", repository = "https://github.com/lsst-ts/ts_ess_labjack" }
+dynamic = [ "version" ]
+
+[tool.setuptools.dynamic]
+version = { attr = "setuptools_scm.get_version" }
+
+[tool.setuptools.packages.find]
+where = [ "python" ]
+
+[project.scripts]
```
Comment (wvreeven) [inline pyproject.toml:19]: Is this section needed at all? Prior to introducing `pyproject.toml` this project didn't have any scripts at all. We don't intend to run this as a stand alone application but only as a "plugin" for the ESS CSC, right?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,129 +1,142 @@
...
+        abortPrevious: true,
+    )
+])
 pipeline {
-
     agent {
-        // Use the docker to assign the Python version.
-        // Use the label to assign the node to run the test.
-        // It is recommended by SQUARE to not add the label
+        // Run as root to avoid permission issues when creating files.
```
Comment (wvreeven) [inline Jenkinsfile:17]: Is this still needed? I thought the docker image we use for the unit tests has all software installed?

**Review by wvreeven — CHANGES_REQUESTED** [review]
In principles this looks good but I have serious doubts whether Jenkins needs to run as root. And there are no scripts in this package so please remove the corresponding section from pyproject.toml.

File: Jenkinsfile (hunk 2 above)
Comment (r-owen) [reply Jenkinsfile:17]: Yes it seems to be. At least I tried to remove it earlier when I was working on updating Eric's template and I could not figure out how to get rid of it. I would love to do so.

**Review by wvreeven — APPROVED** [review]
Looks good to me.

**Pull request merged**